"reliable" mqtt rebased (#121)
ce-rust/cerk: cerk_port_mqtt/README.md

The following configurations are optional.

#### send_qos

The [quality of service](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718099) for message delivery.
The quality of service is only for the MQTT broker and does not change any behavior of the router or the port.

* 0: At most once delivery (default)
* 1: At least once delivery

> * 2: Exactly once delivery

Reviewer: It makes no sense if the router does not support it in general

E.g. `Config::U8(0)`

#### subscribe_topic

> The value has to by of type `Config::String` and contain the MQTT topic name  which the router should subscribe to.

Reviewer: ```suggestion
The value has to be of type `Config::String` and contains the MQTT topic name to which the router should subscribe to.
```

#### subscribe_qos

The [quality of service](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718099) for the topic subscription.

> The value has to by of type `Config::U8` and currently only `0` (At most once delivery) is supported. (see https://github.com/ce-rust/cerk/issues/71)

Reviewer: ```suggestion
The value has to be of type `Config::U8` and currently only `0` (At most once delivery) is supported. (see https://github.com/ce-rust/cerk/issues/71)
```

## Examples

> * [Generator to MQTT](https://github.com/ce-rust/cerk/tree/master/examples/src/mqtt/)

Reviewer: ```suggestion
* [Generator to MQTT](https://github.com/ce-rust/cerk/tree/master/examples/examples/src/mqtt/)
```

## Examples

* [Generator to MQTT](https://github.com/ce-rust/cerk/tree/master/examples/src/mqtt/)

## Limitations

> * **reliability** this port does not support any `DeliveryGuarantee` other then `BestEffort` and so does never send a `OutgoingCloudEventProcessed` or `IncomingCloudEventProcessed` messages.

Reviewer: before you wrote it supports sending with at least once?
